indexer/indexer.go

```diff
@@ -190,6 +191,60 @@ func ImportFromFile(source string) (*Database, error) {
 	return FromBlockStore(blockStore, rootCid)
 }
 
+// Merge two db in ProllyTree level, but the collection merging is not handling now.(i.e. maybe here exists the case
+// that two collections has same name but with different primary keys)
+func Merge(ctx context.Context, db *Database, other *Database) (*Database, error) {
+	err := db.tree.Merge(ctx, other.tree)
+	if err != nil {
+		return nil, err
+	}
+
+	ch, err := other.blockStore.AllKeysChan(ctx)
```

Comment:
Does merging not include any new blocks in the final tree already? I think it should if not instead of leaving it to this layer in the stack

Reply:
Merging(ProllyTree merge) is in ProllyTree layer, but ipld-prolly-indexer only saves links in ProllyTree（https://github.com/RangerMauve/ipld-prolly-indexer/blob/b83d3ceffc9da73eca10f570f1c039097e9e40cc/indexer/indexer.go#L530). i.e. You can save cids in ProllyTree but the data referenced by cid in other databases

Reply:
Excellent, ty for clarifying. Mind adding a comment that mentions this?